### PR TITLE
 check network reachibitiy on pref update

### LINKED ISSUE #3683: Site Settings: should not allow user to make changes when offline
Steps to reproduce:
1. Go to site settings while online
2. Turn off internet connectivity
3. See that you can change the site title and it gets saved in the UI, even though there's a toast indicating that the change was not saved:
02-editing-site-title
03-couldnt-save-info

I expected not to be able to click on any of the settings while offline (and for the settings to be greyed out), and for the orange no connection indicator to be shown. If I do click on a setting that's usually enabled, I expect a toast indicating why I'm not able to use it (i.e. no internet connectivity).

### PR DESCRIPTION
**Pull request by TETRA2000**
fix  #3683

### CODE REVIEW

**Comment by tonyr59h** [comment]
Thanks for contributing! Your changes definitely solve the problem with updating the preference value :+1:

I think we should also disable all the preferences - like we do with the Address preference - if the user loses connection. [Here](https://github.com/TETRA2000/WordPress-Android/blob/issue-3683/WordPress/src/main/java/org/wordpress/android/ui/prefs/BlogPreferencesActivity.java#L218) is where we receive the connection change event.

**Comment by TETRA2000** [comment]
@tonyr59h Thank you for your review.

I agree with your opinion.
So I updated my patch to disable all the preferences while offline.

Please review again.
Thank you.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java [hunk 1]
```diff
@@ -583,6 +585,25 @@ private void initPreferences() {
         if (!mBlog.isAdmin()) hideAdminRequiredPreferences();
     }
 
+    public void setEditingEnabled(boolean enabled) {
+        // excludes mAddressPref, mMorePreference
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java]: :+1:

**Comment by tonyr59h** [comment]
Looks good! Thanks again for contributing :smile: 
:shipit:

**Comment by TETRA2000** [comment]
@tonyr59h Thank you!!

**Pull request merged**